crates/workspace/src/pane_group.rs

```diff
@@ -1118,10 +1151,19 @@ mod element {
                 child.prepaint_at(origin, window, cx);
 
                 origin = origin.apply_along(self.axis, |val| val + child_size.along(self.axis));
+
+                let is_leaf_pane = self
+                    .is_leaf_pane_mask
+                    .lock()
```

Comment:
I do not understand: here, we have `&mut self`, and `with_is_leaf_pane_mask` requires `self`.

There's no clones and some inter-task/thread state exchange that happens: why do we need `Arc` and `Mutex` here? 
Would it be possible to use `Vec<T>` directly here instead?